cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -106,8 +110,27 @@ class GcsTestbench : public ::testing::Environment {
         continue;
       }
 
-      server_process_ = bp::child(exe_path, "-m", "testbench", "--port", port_, group_);
-      if (server_process_.valid() && server_process_.running()) break;
+      bp::ipstream output;
+      server_process_ = bp::child(exe_path, "-m", "testbench", "--port", port_, group_,
+                                  bp::std_err > output);
+      // Wait for message: "* Restarting with"
+      auto testbench_is_running = [&output, this](bp::child& process) {
+        std::string line;
+        std::chrono::time_point<std::chrono::system_clock> end =
+            std::chrono::system_clock::now() + std::chrono::seconds(10);
```

Comment:
nit: std::chrono::steady_clock instead?

Reply:
All other instances of clock in the file are `system_clock`. Should I keep as is then, or change them all? Or is there a good reason to use stead_clock for this use case?

Reply:
Hmm, so this is a local system timeout -> a monotonic clock is what you want
Those other timeouts are deadlines -> I think those go across the wire (gRPC has deadline propagation) and so need to be an actual time point in order to be comparable across machines

That said it's not a big deal here, hence the nit